### PR TITLE
Fixing code location log level filtering

### PR DESCRIPTION
**Pull request by aquesnel**
As part of the logging refactoring in #1708 I broke the code location based log level filtering. This pull request fixes the code location log level filtering.

### CODE REVIEW

File: common/log.c [hunk 1]
```diff
@@ -532,17 +533,20 @@ internal_log_is_enabled_for_level(const enum logLevels log_level,
...
+               && ((override_destination_level && log_level <= override_log_level)
+                  || (!override_destination_level && log_level <= g_staticLogConfig->log_level)))
            || (g_staticLogConfig->enable_syslog
-               && (override_destination_level || log_level <= g_staticLogConfig->syslog_level))
+               && ((override_destination_level && log_level <= override_log_level)
+                  || (!override_destination_level && log_level <= g_staticLogConfig->syslog_level)))
            || (g_staticLogConfig->enable_console
-               && (override_destination_level || log_level <= g_staticLogConfig->console_level));
+               && ((override_destination_level && log_level <= override_log_level)
+                  || (!override_destination_level && log_level <= g_staticLogConfig->console_level)));
```
Comment (matt335672) [inline common/log.c:543]: I'm a little concerned with this mega-conditional. It's quite hard to parse mentally (at least it is for me), and there's some repetition in it. Maybe split it up a bit? Here's a suggestion:-

```c
{
    bool_t result;
    /* Is log initialized? */
    if (g_staticLogConfig == NULL)
    {
        result = 0;
    }
    else if (override_destination_level)
    {
        /* Override in place - if the message should be logged, is any
         * destination enabled? */
        if (log_level > override_log_level)
        {
            result = 0;
        }
        else
        {
            result =
                g_staticLogConfig->fd >= 0 ||
                g_staticLogConfig->enable_syslog ||
                g_staticLogConfig->enable_console;
        }
    }
    else
    {
        /* Is there at least one log destination which will accept the
         * message based on the log level? */
        result =
            (g_staticLogConfig->fd >= 0 &&
             log_level <= g_staticLogConfig->log_level) ||
            (g_staticLogConfig->enable_syslog &&
             log_level <= g_staticLogConfig->syslog_level) ||
            (g_staticLogConfig->enable_console &&
             log_level <= g_staticLogConfig->console_level);
    }

    return result;
}
```

File: common/log.h [hunk 2]
```diff
@@ -69,66 +70,67 @@ enum logReturns
 #define LOG_PER_LOGGER_LEVEL
 
 /**
- * @brief Logging macro for messages that are for an XRDP developper to 
+ * @brief Logging macro for messages that are for an XRDP developper to
  * understand and debug XRDP code.
- * 
- * Note: all log levels are relavant to help a developper understand XRDP at 
+ *
+ * Note: all log levels are relavant to help a developper understand XRDP at
```
Comment (matt335672) [inline common/log.h:76]: Suggest replace 'developper' with 'developer'

**Review by matt335672 — COMMENTED** [review]
Some minor comments

File: common/log.c (hunk 1 above)
Comment (aquesnel) [reply common/log.c:543]: cleaned up to be easier to understand.

File: common/log.h (hunk 2 above)
Comment (aquesnel) [reply common/log.h]: fixed

**Comment by aquesnel** [comment]
@matt335672 I've implemented the changes you suggested.

Please let me know if more changes are needed before merging.

**Comment by metalefty** [comment]
Is this should be in v0.9.15?

**Comment by aquesnel** [comment]
@metalefty Yes I think this pull request should be in v0.9.15 because the log filtering by source location in devel is broken without this pull request.

File: common/log.c [hunk 3]
```diff
@@ -530,17 +530,37 @@ internal_log_is_enabled_for_level(const enum logLevels log_level,
...
-               && ((override_destination_level && log_level <= override_log_level)
-                  || (!override_destination_level && log_level <= g_staticLogConfig->console_level)));
+    else if (g_staticLogConfig->fd < 0
+            && !g_staticLogConfig->enable_syslog
+            && !g_staticLogConfig->enable_console)
+    {
+        /* all logging outputs are disabled */
+        return 0;
+    }
+    else if (override_destination_level)
```
Comment (matt335672) [inline common/log.c:540]: I'm happy with either multiple returns in a function, or if / else if with a single return at the end. Here you've mixed styles somewhat. It'll work fine, but I think there's a chance that in the future a static analysis will moan about an unreachable 'else' branch following a return. 

I'd suggest either removing the unnecessary `else` statements, or using a single return - whatever seems more natural to you.

Comment (aquesnel) [reply common/log.c:541]: @matt335672 I've implemented this feedback using multiple returns by only have a single level of nesting for the if/else. Please let me know if I correctly interpreted your feedback, or if you'd prefer that I use the single return at the end style.

**Comment by aquesnel** [comment]
CR feedback implemented, branch squashed and rebased to devel.

This should be ready to merge once @matt335672 reviews the changes.

**Review by metalefty — APPROVED** [review]

File: common/log.c (hunk 3 above)
Comment (matt335672) [reply common/log.c:541]: Sorry - I've probably not been very clear.

My point was that in this kind of construction:-

<pre>
if (&lt;something&gt;)
{
    return 1;
}
<b>else</b> if (&lt;something else&gt;)
...
</pre>

the `else` in bold is superfluous and not required. At some stage I'd expect a static analysis tool to say something about it. Does that make more sense?

The nesting style you use, or whether you use a single return or multiple returns is completely up to you. I've got my own preferences but they're not relevant at all.
